In [4]:
import numpy as np
import pandas as pd

In [5]:
ds = pd.read_csv("Titan_Sales.csv")
ds["Old Scheme"] = ds["Old Scheme (in thousands)"] * 1000 
ds["New Scheme"] =ds ["New Scheme (in thousands)"] * 1000
ds.describe()

ALESPERSON  Old Scheme (in thousands)  New Scheme (in thousands)  \
count   30.000000                  30.000000                  30.000000   
mean    15.500000                  68.033333                  72.033333   
std      8.803408                  20.455980                  24.062395   
min      1.000000                  28.000000                  32.000000   
25%      8.250000                  54.000000                  55.000000   
50%     15.500000                  67.000000                  74.000000   
75%     22.750000                  81.500000                  85.750000   
max     30.000000                 110.000000                 122.000000   

          Old Scheme     New Scheme  
count      30.000000      30.000000  
mean    68033.333333   72033.333333  
std     20455.980212   24062.394947  
min     28000.000000   32000.000000  
25%     54000.000000   55000.000000  
50%     67000.000000   74000.000000  
75%     81500.000000   85750.000000  
max    110000.000000  122000.000000

# Answer 1 

In [6]:
ds["Old Scheme"].mean()

68033.33333333333

In [7]:
ds["New Scheme"].mean()


72033.33333333333

In [8]:
diffmean = ds["New Scheme"].mean() -ds["Old Scheme"].mean()
diffmean

4000.0

# Answer 2

mu1 = Average sums assured by salesperson in old scheme.
mu2 = Average sums assured by salesperson in new scheme.

H0: mu1 = mu2;There is no significant difference in outputs 
H1: m2 - m1  > 0; There is significant difference in outputs

alpha= .05

In [9]:
import scipy.stats as stats
s,p = stats.ttest_rel(ds["Old Scheme"],ds["New Scheme"])
s,p

(-1.5559143823544377, 0.13057553961337662)

In [10]:
p <=0.05


False

pvalue > alpha ,hence accept null hypothesis.The new scheme has not raised outputs signifiantly.

# Answer 3

A small p-value (typically ≤ 0.05) indicates strong evidence against the null hypothesis, so you reject the null hypothesis.

A large p-value (> 0.05) indicates weak evidence against the null hypothesis, so you fail to reject the null hypothesis.


# Answer 4(a)

When the null hypothesis is true and you reject it, you make a type I error. The probability of making a type I error is alpha, which is the level of significance you set for your hypothesis test.
P(Type1 Error) = 0.05


# Answer 4(b)

m2 = Average sums assured by salesperson in new scheme.

m1 = Average sums assured by salesperson in old scheme

H0: m2-m1 ≤ 5000 (statistially)  H1: m2-m1 > 5000
        

In [11]:
from numpy import mean
from math import sqrt
from scipy.stats import t
def dependent_ttest(data1, data2, alpha,meanDif):
	n = len(data1)
	d1 = sum([(data1[i]-data2[i])**2 for i in range(n)])
	d2 = sum([data1[i]-data2[i] for i in range(n)])
	sd = sqrt((d1 - (d2**2 / n)) / (n - 1))
	sed = sd / sqrt(n)
	t_stat = (meanDif) / sed
	df = (n - 1)
	cv = t.ppf(1.0 - alpha, df)
	p = (1.0 - t.cdf(abs(t_stat), df)) * 2.0
	return t_stat, cv, p,sed
t_stat, cv, p,sed=dependent_ttest(ds['New Scheme'],ds['Old Scheme'],0.05,5000)
t_stat, cv, p,sed

(1.9448929779430468,
 1.6991270265334972,
 0.06154216863016693,
 2570.8355455569017)

#Answer 4(c) 

m2 = Average sums assured by salesperson in new scheme.

m1 = Average sums assured by salesperson in old scheme
H0: m2-m1 = 4000 (statistially)  H1: m2-m1 >= 0

H0 will be rejected if test statistics > t_critical.

alpha = 0.05 
degree of Freedom = 29,
critical value for t statistic  will be 1.699.



In [12]:
xbar= cv*sed
xbar

4368.176156228719

Hence, H0 will be rejected when xbar ≥ 4368.17

Probability(Type 2 error) 
=P (xbar < 4368.176)

In [13]:
t1 = (xbar-5000)/sed
p =1- stats.t.cdf(t1,29)
p

0.5962027475205683

In [14]:
p =1- stats.t.cdf(t1,29)
p

0.5962027475205683

Power of test = 1-p

In [15]:
1-p

0.40379725247943166